readthedocs_ext/readthedocs.py

```diff
@@ -249,9 +249,12 @@ def remove_search_init(app, exception):
         replacement_text = '/* Search initialization removed for Read the Docs */'
         replacement_regex = re.compile(
             r'''
-            ^\$\(document\).ready\(function\s*\(\)\s*{(?:\n|\r\n?)
+            ^(\$\(document\).ready\(function\s*\(\)\s*{(?:\n|\r\n?)
             \s*Search.init\(\);(?:\n|\r\n?)
-            \}\);
+            \}\);)
+            |
+            # Sphinx >=5.0 calls Search.init this way.
+            (_ready\(Search.init\);)
             ''',
```

Comment:
I think it would be good to add links to the sphinx code we are expecting to search & replace as comments in this regex for easier tracking/understanding of it.

* https://github.com/sphinx-doc/sphinx/blob/799385f5558a888d1a143bf703d06b66d6717fe4/sphinx/themes/basic/static/searchtools.js#L527-L529
* https://github.com/sphinx-doc/sphinx/blob/3b01fbe2adf3077cad5c2cf345c6d000d429d7ac/sphinx/themes/basic/static/searchtools.js#L507

Reply:
@stsewd it seems you missed this comment.

Reply:
I updated the docstring https://github.com/readthedocs/readthedocs-sphinx-ext/pull/100/commits/ad92c762d4a06a8c55b54df42e3885b28dc4ab7e

Reply:
Oh, I see. I missed that :)